caas/kubernetes/provider/application/application.go

```diff
@@ -143,34 +142,52 @@ func (a *app) Ensure(config caas.ApplicationConfig) (err error) {
 	}
 	applier.Apply(&secret)
 
-	if len(charmDeployment.ServicePorts) > 0 {
-		service := resources.Service{
-			Service: corev1.Service{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:        a.name,
-					Namespace:   a.namespace,
-					Labels:      a.labels(),
-					Annotations: a.annotations(config),
-				},
-				Spec: corev1.ServiceSpec{
-					Selector: a.labels(),
-					Type:     corev1.ServiceTypeClusterIP,
+	/*
+		if len(charmDeployment.ServicePorts) > 0 {
+			service := resources.Service{
+				Service: corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        a.name,
+						Namespace:   a.namespace,
+						Labels:      a.labels(),
+						Annotations: a.annotations(config),
+					},
+					Spec: corev1.ServiceSpec{
+						Selector: a.labels(),
+						Type:     corev1.ServiceTypeClusterIP,
+					},
 				},
-			},
-		}
+			}
 
-		for _, v := range charmDeployment.ServicePorts {
-			port := corev1.ServicePort{
-				Name:       v.Name,
-				Port:       int32(v.Port),
-				TargetPort: intstr.FromInt(v.TargetPort),
-				Protocol:   corev1.Protocol(v.Protocol),
-				//AppProtocol:    core.Protocol(v.AppProtocol),
+			for _, v := range charmDeployment.ServicePorts {
+				port := corev1.ServicePort{
+					Name:       v.Name,
+					Port:       int32(v.Port),
+					TargetPort: intstr.FromInt(v.TargetPort),
+					Protocol:   corev1.Protocol(v.Protocol),
+					//AppProtocol:    core.Protocol(v.AppProtocol),
+				}
+				service.Spec.Ports = append(service.Spec.Ports, port)
 			}
-			service.Spec.Ports = append(service.Spec.Ports, port)
+			applier.Apply(&service)
 		}
-		applier.Apply(&service)
+	*/
+
+	service := resources.Service{
+		Service: corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        a.name,
+				Namespace:   a.namespace,
+				Labels:      a.labels(),
+				Annotations: a.annotations(config),
+			},
+			Spec: corev1.ServiceSpec{
+				Selector: a.labels(),
+				Type:     corev1.ServiceTypeClusterIP,
+			},
+		},
```

Comment:
We probably don't want to revert an already port opened svc to the initial version in Ensure here.
So the logic here should create first, and ignore the already exist error.
I am currently working on service stuff, so we can leave it as it is for now.

Reply:
Yep agreed, I'll leave this to you.